syntax = "proto3";

// Fade multiple light channels in the platform
// This might be too low level for VPE as a RGB light would be three channels and a GI only one
// We might want to separate RGB and single color lights (or handle single color lights as RGB as well)
message FadeLightRequest {
  message ChannelFade {
    string light_number = 1;
    float target_brightness = 2;
  }
  uint32 common_fade_ms = 1;
  repeated ChannelFade fades = 2;
}

// Alternative for RGB
message FadeRGBRequest {
  string light_number = 1;
  // or: string target_hex_color = 2;
  float target_brightness_r = 2;
  float target_brightness_g = 3;
  float target_brightness_b = 4;
  uint32 fade_ms = 5;
}

// Alternative for GI/single color lights
message FadeSingleColorRequest {
  string light_number = 1;
  float target_brightness = 2;
  uint32 fade_ms = 3;
}

message FadeLightResponse {
}
